lms/resources/lti_launch.py

```diff
@@ -29,12 +30,7 @@ def __init__(self, request):
     @cached_property
     def course(self):
         """Get the course this LTI launch based on the request's params."""
-
-        return self._request.find_service(name="course").upsert_course(
-            context_id=self._request.parsed_params["context_id"],
-            name=self._request.parsed_params["context_title"],
-            extra=self._course_extra(),
-        )
+        return self._request.find_service(LTILaunchService).record_course()
```

Comment:
There's still a couple of places where this is used in JSConfig. I'd aim to remove it but I want to do that in a different refactor otherwise this one will get huge.